interpreter/cling/lib/Interpreter/NullDerefProtectionTransformer.cpp

```diff
@@ -91,32 +91,45 @@ namespace cling {
     Expr* SynthesizeCheck(Expr* Arg) {
       assert(Arg && "Cannot call with Arg=0");
 
-      if(!m_LookupResult)
+      if(!m_clingthrowIfInvalidPointerCache)
         FindAndCacheRuntimeLookupResult();
 
       SourceLocation Loc = Arg->getLocStart();
-
       Expr* VoidSemaArg = utils::Synthesize::CStyleCastPtrExpr(&m_Sema,
                                                             m_Context.VoidPtrTy,
                                                             (uint64_t)&m_Sema);
       Expr* VoidExprArg = utils::Synthesize::CStyleCastPtrExpr(&m_Sema,
                                                           m_Context.VoidPtrTy,
                                                           (uint64_t)Arg);
-
-      Expr *args[] = {VoidSemaArg, VoidExprArg, Arg};
-
       Scope* S = m_Sema.getScopeForContext(m_Sema.CurContext);
       CXXScopeSpec CSS;
-      Expr* unresolvedLookup
-        = m_Sema.BuildDeclarationNameExpr(CSS, *m_LookupResult,
-                                         /*ADL*/ false).get();
-      Expr* call = m_Sema.ActOnCallExpr(S, unresolvedLookup, Loc,
-                                        args, Loc).get();
-      TypeSourceInfo* TSI
-              = m_Context.getTrivialTypeSourceInfo(Arg->getType(), Loc);
-      Expr* castExpr = m_Sema.BuildCStyleCastExpr(Loc, TSI, Loc, call).get();
 
-      return castExpr;
+      Expr* clingThrowIfInvalidPointerCallExpr
```

Comment:
Here it's okay to name it "checkCall" or similar. No need to be *that* verbose ;-)